File: server/api/src/main/java/main/entities/Restaurante/Empleado.java
package main.entities.Restaurante;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;
import main.entities.Domicilio.Domicilio;

import java.time.LocalDate;
import java.util.HashSet;
import java.util.Set;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Builder
@Table(name = "empleados", schema = "buen_sabor")
public class Empleado {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @Column(name = "nombre")
    private String nombre;
    @Column(name = "email")
    private String email;
    @Column(name = "contraseña")
    private String contraseña;
    @Column(name = "cuil")
    private String cuil;
    @Column(name = "telefono")
    private Long telefono;

    @JsonIgnoreProperties(value = {"domicilios"}, allowSetters = true)
    @OneToMany(fetch = FetchType.LAZY, mappedBy = "empleado", cascade = CascadeType.ALL)
    private Set<Domicilio> domicilios = new HashSet<>();

    @JsonIgnoreProperties(value = {"empleado"}, allowSetters = true)
    @OneToMany(mappedBy = "empleado", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER)
    private Set<FechaContratacionEmpleado> fechaContratacion = new HashSet<>();

    @Column(name = "fecha_nacimiento")
    private LocalDate fechaNacimiento;

    @Column(name = "borrado")
    private String borrado = "NO";

    @JsonIgnoreProperties(value = {"empleado"}, allowSetters = true)
    @OneToMany(mappedBy = "empleado", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<EmpleadoPrivilegio> empleadoPrivilegios = new HashSet<>();

    @JsonIgnoreProperties(value = {"empleados", "empresa", "stocksIngredientes", "stocksArticulo", "promociones", "localidadesDisponiblesDelivery", "articulosMenu", "articulosVenta", "medidas", "categorias", "imagenes", "ingredientes", "stocksEntranteSucursal"}, allowSetters = true)
    @ManyToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinTable(
            name = "empleados_sucursales",
            joinColumns = @JoinColumn(name = "id_empleado"),
            inverseJoinColumns = @JoinColumn(name = "id_sucursal")
    )
    private Set<Sucursal> sucursales = new HashSet<>();

}

File: src/types/Restaurante/Empleado.ts
import { Domicilio } from "../Domicilio/Domicilio";
import { PrivilegiosEmpleados } from "./PrivilegiosEmpleado";
import { FechaContratacionEmpleado } from "./FechaContratacionEmpleado";
import { RolesEmpleado } from "./RolesEmpleados";
import { Sucursal } from "./Sucursal";
import { Imagenes } from "../Productos/Imagenes";

export class Empleado {
    id: number = 0;
    nombre: string = '';
    email: string = '';
    contraseña: string = '';
    cuil: string = '';
    telefono: number = 0;
    domicilios: Domicilio[] = [];
    fechaContratacion: FechaContratacionEmpleado[] = [];
    fechaNacimiento: Date = new Date();
    privilegios: PrivilegiosEmpleados[] = [];
    sucursales: Sucursal[] = [];
    rolesEmpleado: RolesEmpleado[] = [];
    borrado: string = '';
    imagenes: Imagenes[] = [];
    
    constructor() {

    }
}